test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/reliefdetails/ReliefDetailsControllerSpec.scala

```diff
@@ -1350,6 +1355,39 @@ object ReliefDetailsControllerSpec extends Matchers {
     reliefDetailsAnswers: CompleteReliefDetailsAnswers,
     doc: Document
   )(implicit messages: MessagesApi, lang: Lang): Unit = {
-    // TODO Implement checks
+
+    if (reliefDetailsAnswers.privateResidentsRelief.isZero) {
+      doc.select("#privateResidentsRelief-answer").text shouldBe "No"
+    } else {
+      doc.select("#privateResidentsRelief-answer").text shouldBe "Yes"
+      doc.select("#privateResidentsReliefValue-answer").text shouldBe formatAmountOfMoneyWithPoundSign(
+        reliefDetailsAnswers.privateResidentsRelief.inPounds()
+      )
+    }
+
+    if (reliefDetailsAnswers.lettingsRelief.isZero) {
+      doc.select("#lettingsRelief-answer").text shouldBe "No"
+    } else {
+      doc.select("#lettingsRelief-answer").text shouldBe "Yes"
+      doc.select("#lettingsReliefValue-answer").text shouldBe formatAmountOfMoneyWithPoundSign(
+        reliefDetailsAnswers.lettingsRelief.inPounds()
+      )
+    }
+
+    reliefDetailsAnswers.otherReliefs.getOrElse(NoOtherReliefs) match {
+      case a: OtherReliefsOption.OtherReliefs =>
+        doc.select("#otherReliefs-answer").text shouldBe "Yes"
+        doc.select("#otherReliefsName-answer").text shouldBe new String(
+          reliefDetailsAnswers.otherReliefs.getOrElse[OtherReliefsOption](sys.error("Error")) match {
+            case a: OtherReliefs => a.name.toString
+          }
+        )
+        doc.select("#otherReliefsAmount-answer").text shouldBe formatAmountOfMoneyWithPoundSign(
+          reliefDetailsAnswers.otherReliefs.getOrElse[OtherReliefsOption](sys.error("Error")) match {
+            case a: OtherReliefs => a.amount.inPounds().bigDecimal
+          }
+        )
+      case _ => if (reliefDetailsAnswers.otherReliefs.isDefined) doc.select("#otherReliefs-answer").text shouldBe "No"
```

Comment:
oh, I see what you've done. You could just do something like

 ```scala
reliefDetailsAnswers.otherReliefs.foreach{
  case a: OtherReliefsOption.OtherReliefs => ...

  case n: OtherReliefsOption.NoOtherReliefs => ...
}
```
Then checks are only done when the option is defined, and no checks are done if it's not.